store/sdk/src/main/java/org/apache/carbondata/sdk/file/Field.java

```diff
@@ -80,6 +82,24 @@ public Field(String name, String type) {
       this.type = DataTypes.DOUBLE;
     } else if (type.equalsIgnoreCase("binary")) {
       this.type = DataTypes.BINARY;
+    } else if (type.startsWith("decimal")) {
```

Comment:
Please make it as type.toLowerCase.startsWith(...) to handle case sensitivity.

Reply:
done